schema/resource.go

```diff
@@ -91,6 +95,41 @@ func (r *Resource) Columns() []string {
 	return r.Table.Columns.Names()
 }
 
+//nolint:revive
+func (r *Resource) CalculateUniqueValue(consistentID bool) error {
```

Comment:
Is there a reason this couldn't be implemented as a resolver?

Reply:
When making cq_id deterministic we cannot use any of the existing resolver types because we cannot guarantee that the value has been fetched by that time so in this PR I call a new function after the resource has been fully resolved in resolveResource